contracts/golos.vesting/golos.vesting.cpp

```diff
@@ -0,0 +1,500 @@
+            eosio_assert(balance != account_recipient.end(), "This token is not on the recipient balance sheet");
+            account_recipient.modify(balance, sender, [&](auto &item){
+                item.received_vesting -= quantity;
+            });
+
+            tables::account_table account_sender(_self, sender);
+            auto balance_sender = account_sender.find(quantity.symbol.name());
+            eosio_assert(balance_sender != account_sender.end(), "This token is not on the sender balance sheet");
+            account_sender.modify(balance_sender, sender, [&](auto &item){
+                item.delegate_vesting -= quantity;
+            });
+
+            flag_recipient = true;
+            break;
+        }
+
+        ++it_index;
+    }
+
+    eosio_assert(flag_recipient, "Error undeligate, not found recipient");
+}
+
+void vesting::calculate_convert_vesting() {
+    require_auth(_self);
+
+    tables::vesting_table table_pair(_self, _self);
+    for(auto obj_pair : table_pair) {
+        tables::convert_table table(_self, obj_pair.vesting.symbol.name());
+        auto index = table.get_index<N(payout_time)>();
+        for (auto obj = index.cbegin(); obj != index.cend(); ) {
+            if (obj->payout_time <= time_point_sec(now())) {
+                if(obj->number_of_payments > 0) {
+                    index.modify(obj, 0, [&](auto &item) {
+                        item.payout_time = time_point_sec(now() + OUTPUT_PAYOUT_PERIOD);
+                        --item.number_of_payments;
+
+                        print(uint64_t(item.number_of_payments));
+
+                        tables::account_table account(_self, obj->sender);
+                        auto balance = account.find(obj->payout_part.symbol.name());
+
+                        auto lambda_action_sender = [&](asset quantity){
+                            account.modify(balance, 0, [&](auto &item){
+                                item.vesting -= quantity;
+                            });
+
+                            auto sym_name = quantity.symbol.name();
+                            auto index = table_pair.get_index<N(vesting)>();
+                            auto pair = index.find(sym_name);
+                            eosio_assert(pair != index.end(), "Vesting not found");
+
+                            index.modify(pair, 0, [&](auto &item) {
+                                item.vesting -= quantity;
+                                item.token -= convert_token(quantity, pair->token.symbol);
+                            });
+
+                            INLINE_ACTION_SENDER(eosio::token, transfer)( N(eosio.token), {_self, N(active)},
+                            { _self, obj->recipient, convert_token(quantity, pair->token.symbol), "Translation into tokens" } );
+                        };
+
+                        if (balance->vesting < obj->payout_part) {
+                            item.number_of_payments = NOT_TOKENS_WEEK_PERIOD;
```

Comment:
In this case, the smart contract must convert the existing part and stop the converting.